### DISCUSSION TITLE
Suggested modification of sample code: "listen in a thread"

### DISCUSSION THREAD

**Discussion by ywshzh** [python-smpplib/python-smpplib#149] [Ideas]
`
from threading import Thread
t = Thread(target=client.listen)
t.start()
`
I used the code in the example to implement the function of sending SMS in django, but when listen() throws an exception, the thread terminates and the enquire_link packet loss, it will causes the server unbind.
My uwsgi server threads are set to 4, and 4 threads constantly rebind the server, it will resulting whith similar errors like: smpplib. exceptions. UnknownCommand Error: Unknown SMPP command code "0x428"

So，would the following modifications be better?
`
def listen():
	while True:
		try:
			ecarxsmpp.client.listen()
		except Exception:
			print traceback.format_exc()

t = Thread(target=listen)
t.start()
`

**Comment by code-of-kpp** [comment]
I think most of the people just wrap `listen()` code in some kind of try-catch block. And when some exception is caught they signal main thread about it. I think the only reasonable thing to do here is just reestablish the connection and restart `listen()`: either by signaling existing thread back or by restarting (stop-create-start) listening thread.

**Comment by ywshzh** [comment]
Thank you for your prompt reply.
